custom/covid/management/commands/update_all_activity_complete_date.py

```diff
@@ -51,19 +62,107 @@ def handle(self, domain, **options):
             )
 
 
-def _get_bad_case_ids(domain):
-    return list(CaseSearchES()
-                .domain(domain)
-                .case_property_query("all_activity_complete_date", "date(today())")
-                .modified_range(gte=datetime.date(2022, 2, 2))
-                .case_type(["patient", "contact"])
-                .scroll_ids())
+def _get_bad_case_ids(domain, case_type):
+    query = (CaseSearchES()
+             .domain(domain)
+             .filter(exact_case_property_text_query('current_status', 'closed'))
+             .filter(case_property_missing("all_activity_complete_date"))
+             .case_type(case_type)
+             .owner(INACTIVE_LOCATION_IDS)
+             .modified_range(gte=datetime.date(2022, 2, 2)))
+
+    if case_type == 'contact':
+        query = query.NOT(
+            exact_case_property_text_query('finale_disposition', 'converted_to_pui')
+        )
+
+    return list(query.scroll_ids())
 
 
 def _correct_bad_property(case):
-    if case.get_case_property('all_activity_complete_date') == 'date(today())':
-        return CaseBlock(
-            create=False,
-            case_id=case.case_id,
-            update={"all_activity_complete_date": ""},
-        )
+    if (
+            # Double check filters in case ES data is stale
+            case.get_case_property('all_activity_complete_date')
+            or case.get_case_property('current_status') != 'closed'
+            or (case.type == 'contact'
+                and case.get_case_property('finale_disposition') == 'converted_to_pui')
+            or case.owner_id not in INACTIVE_LOCATION_IDS
+    ):
+        return None
+
+    fallback_property = 'quarantine_end_date' if case.type == 'contact' else 'isolation_end_date'
+    new_value = case.get_case_property(fallback_property) or '2022-02-02'
+    return CaseBlock(
+        create=False,
+        case_id=case.case_id,
+        update={"all_activity_complete_date": new_value},
+    )
+
+
+INACTIVE_LOCATION_IDS = [
```

Comment:
Michel sent me this [lookup table](https://www.commcarehq.org/a/ny-dev-cdcms/fixtures/view_lookup_tables/?table_id=b6f354cd58064b2c9e18bd69f3910849), but since that's updated manually, I figure a hardcoded list from that would be fine for a short-lived script.  I'd like to lookup locations on each domain where `cict_location_type = 'inactive'`, but I wasn't certain how reliable that is.